### PR TITLE
Add upstream PR Review/Contribute Practices

### PR DESCRIPTION
**Pull request by chunfuwen**
This is follow-up for issue:#2748
Libvirt Team members summarize some rules or guidelines related to Upstream PR Review/Contribute as per past practices.
After reviewing, it is good to be merged and become common best practices in avocado-vt upstream.
If further feedbacks are available, related guidelines and rules can be constantsly evolved accordingly.

Signed-off-by: chunfuwen <chwen@redhat.com>

### CODE REVIEW

**Comment by chunfuwen** [comment]
This is a follow-up for issue :https://github.com/avocado-framework/avocado-vt/issues/2748

**Comment by pevogam** [comment]
I think this is best discussed after #2794 where we will reenable the vital parts of the CI and could add linters on top automating some of these requirements once properly considered.

**Comment by chunfuwen** [comment]
> I think this is best discussed after #2794 where we will reenable the vital parts of the CI and could add linters on top automating some of these requirements once properly considered.

Sorry, I am not sure how #2794 is linked with this PR.

**Comment by pevogam** [comment]
The main reason was that we have to see if everything works after allowing some minimal checks for the CI which also check styles and best practices. In fact I am confused - shouldn't the #2794 be closed now considering its PR is already merged?

**Comment by chunfuwen** [comment]
> The main reason was that we have to see if everything works after allowing some minimal checks for the CI which also check styles and best practices. In fact I am confused - shouldn't the #2794 be closed now considering its PR is already merged?

issue # 2794 was already addressed by merged PR:https://github.com/avocado-framework/avocado-vt/pull/2797 ,so I close it.

File: PR_Review_And_Contribute_Practices.txt [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+7)Remember that there is often more than one way to approach a solution.
+8)Given clear and useful comments, and explain the reason why reqest change
+9)In general, reviewers should favor approving a PR once it is in a state where it definitely improves the overall code health of the system being worked on,
+  even if the PR isn't perfect.(maintainability, readability, and understandability)
+10)Share your best practice/knowledge as a mentor.
+11)Cautiously regard personal preference as best practice and impose to contributors
+12)Dismiss your approval if you add new comment for a PR after you already have given an approval
+13)It is ok to use a 'request review' tool to ask someone to review as you like, but not a must.
+14)It is ok to cancel the request review to you if you think you are not a suitable one for this PR.
+15)Wait for request review if there is for 2 weeks for those PRs which have already 2 approvals
```
Comment (kylazhang) [inline PR_Review_And_Contribute_Practices.txt:21]: I think the description for No.15 is not so clear to understand

**Comment by dzhengfy** [comment]
@chunfuwen Could you help refine the commit message before we merge it? For example, remove below useless messages:
```
Signed-off-by: chunfuwen <chwen@redhat.com>
test
```

File: PR_Review_And_Contribute_Practices.txt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+2)[Must] PR commit message is meaningful. Refer to the link on how to write a good commit message
+3)[Must] Travis CI pass and no conflict
+4)[Must] Provide test results. If no, provide justification. Apply to any PR
+  One of below options should be aligned:
+5)[Must] If the function defined with right docstring (description and params, and return if have)
+6)[Must] If the PR depends on other PRs, ensure the PR is merged after other PRs
+7)[Must] If the API of one library is changed, ask for all test cases to be modified which invoke this library and provide test results of representative test cases.
+8)[Must] If the case does some package version judgement for the new case support or compatible backwards
+9)[Must] If the test code have suitable env backup and recovery steps
+10)[Must] Add a comment to say if your PR has a dependence
```
Comment (kylazhang) [inline PR_Review_And_Contribute_Practices.txt:49]: duplicate with No.6

File: PR_Review_And_Contribute_Practices.txt (hunk 1 above)
Comment (chunfuwen) [reply PR_Review_And_Contribute_Practices.txt]: updated with :
Wait for request review for no more than 2 weeks on those PRs which have already 2 approvals

File: PR_Review_And_Contribute_Practices.txt (hunk 2 above)
Comment (chunfuwen) [reply PR_Review_And_Contribute_Practices.txt]: merged with No.6

**Comment by chunfuwen** [comment]
> @chunfuwen Could you help refine the commit message before we merge it? For example, remove below useless messages:
> 
> ```
> Signed-off-by: chunfuwen <chwen@redhat.com>
> test
> ```

thanks, updated

**Review by dzhengfy — APPROVED** [review]
LGTM

**Review by kylazhang — APPROVED** [review]

**Pull request merged**